src/sentry_core.c

```diff
@@ -25,6 +25,7 @@
 
 static sentry_options_t *g_options = NULL;
 static sentry_mutex_t g_options_lock = SENTRY__MUTEX_INIT;
+static sentry_mutex_t g_init_lock = SENTRY__MUTEX_INIT;
```

Comment:
I think we should re-use the `g_options_lock` here, so no other thread can touch `g_options` in the middle.

Reply:
Just for readability I would rather keep the additional mutex to see that it guards more than only the access to the data object.